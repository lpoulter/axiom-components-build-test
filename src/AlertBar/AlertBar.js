import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import AlertIcon from '../AlertIcon/AlertIcon';
import Base from '../Base/Base';
import Grid from '../Grid/Grid';
import GridCell from '../Grid/GridCell';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import './AlertBar.css';

export default class AlertBar extends Component {
  static propTypes = {
    /** Content displayed next to the AlertIcon */
    children: PropTypes.node.isRequired,
    /** An optional callback that when given adds a removable cross */
    onRemoveClick: PropTypes.func,
    /** Size of the AlertBar */
    size: PropTypes.oneOf(['small', 'medium']),
    /** Type of AlertBar that affects the coloring and icon */
    type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  };

  static defaultProps = {
    size: 'small',
    type: 'info',
  };

  render() {
    const { children, size, type, onRemoveClick, ...rest } = this.props;
    const classes = classnames('ax-alert-bar', `ax-alert-bar--${size}`, `ax-alert-bar--${type}`);

    return (
      <Base { ...rest } className={ classes }>
        <Grid gutters="tiny" responsive={ false } verticalAlign="middle">
          <GridCell shrink>
            <AlertIcon
                style="subtle"
                theme="day"
                type={ type } />
          </GridCell>

          <GridCell textColor="night" textStrong>
            { children }
          </GridCell>

          { onRemoveClick && (
            <GridCell shrink>
              <Link onClick={ onRemoveClick } style="night">
                <Icon name="cross" />
              </Link>
            </GridCell>
          ) }
        </Grid>
      </Base>
    );
  }
}
