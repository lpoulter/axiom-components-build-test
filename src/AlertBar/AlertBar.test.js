import React from 'react';
import renderer from 'react-test-renderer';
import AlertBar from './AlertBar';

const getComponent = (props = {}) =>
  renderer.create(
    <AlertBar { ...props }>Lorem</AlertBar>
  );

describe('AlertBar', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onRemoveClick', () => {
    const component = getComponent({ onRemoveClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with size', () => {
    ['small', 'medium'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with type', () => {
    ['success', 'warning', 'error', 'info'].forEach((type) => {
      it(type, () => {
        const component = getComponent({ type });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

});
