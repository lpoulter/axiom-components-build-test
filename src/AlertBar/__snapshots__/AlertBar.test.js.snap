// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBar renders with defaultProps 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--info"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--info ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with onRemoveClick 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--info"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--info ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <a
          className="ax-link ax-link--style-night"
          onClick={[Function]}
        >
          <svg
            className="ax-icon ax-icon--cross"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with size medium 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--medium ax-alert-bar--info"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--info ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with size small 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--info"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--info ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with type error 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--error"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--error ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--warning"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with type info 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--info"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--info ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with type success 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--success"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--success ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--tick"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBar renders with type warning 1`] = `
<div
  className="ax-alert-bar ax-alert-bar--small ax-alert-bar--warning"
>
  <div
    className="ax-grid__container ax-space--x6"
  >
    <div
      className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle ax-grid--wrap"
    >
      <div
        className="ax-grid__cell ax-grid__cell--shrink"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="ax-alert-icon ax-alert-icon--subtle ax-alert-icon--small ax-alert-icon--warning ax-theme--day"
        >
          <svg
            className="ax-icon ax-icon--warning"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-grid__cell ax-text--color-night ax-text--strong"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;
